GVRf/Framework/src/org/gearvrf/GVRScene.java

```diff
@@ -169,51 +170,73 @@ public boolean getStatsEnabled() {
 
     /**
      * Set whether to enable display of stats for this scene.
-     *
+     * 
      * @param enabled
-     *     Flag to indicate whether to enable display of stats.
+     *            Flag to indicate whether to enable display of stats.
      */
     public void setStatsEnabled(boolean enabled) {
         pendingStats = enabled;
     }
 
     void updateStatsEnabled() {
-        if(mStatsEnabled == pendingStats) {
+        if (mStatsEnabled == pendingStats) {
             return;
         }
 
         mStatsEnabled = pendingStats;
-        if(mStatsEnabled && mStatsConsole == null) {
-            mStatsConsole = new GVRConsole(getGVRContext(), GVRConsole.EyeMode.BOTH_EYES);
+        if (mStatsEnabled && mStatsConsole == null) {
+            mStatsConsole = new GVRConsole(getGVRContext(),
+                    GVRConsole.EyeMode.BOTH_EYES);
             mStatsConsole.setCanvasWidthHeight(512, 512);
             mStatsConsole.setXOffset(125.0f);
             mStatsConsole.setYOffset(125.0f);
         }
 
-        if(mStatsEnabled && mStatsConsole != null) {
+        if (mStatsEnabled && mStatsConsole != null) {
             mStatsConsole.setEyeMode(GVRConsole.EyeMode.BOTH_EYES);
-        } else if(!mStatsEnabled && mStatsConsole != null) {
+        } else if (!mStatsEnabled && mStatsConsole != null) {
             mStatsConsole.setEyeMode(GVRConsole.EyeMode.NEITHER_EYE);
         }
     }
 
     void resetStats() {
         updateStatsEnabled();
-        if(mStatsEnabled) {
+        if (mStatsEnabled) {
             mStatsConsole.clear();
             NativeScene.resetStats(getNative());
         }
     }
 
     void updateStats() {
-        if(mStatsEnabled) {
+        if (mStatsEnabled) {
             int numberDrawCalls = NativeScene.getNumberDrawCalls(getNative());
             int numberTriangles = NativeScene.getNumberTriangles(getNative());
 
             mStatsConsole.writeLine("Draw Calls: %d", numberDrawCalls);
-            mStatsConsole.writeLine(" Triangles: %d", numberTriangles);
+            mStatsConsole.writeLine("Triangles: %d", numberTriangles);
+
+            if (mStatMessage.length() > 0)
+                mStatsConsole.writeLine("%s", mStatMessage.toString());
         }
     }
+
+    /**
+     * Add an additional string to stats message for this scene.
+     * 
+     * @param message
+     *            String to add to stats message.
+     */
+    public void addStatMessage(String message) {
+        mStatMessage = new StringBuffer(message);
+    }
```

Comment:
This will be fixed in a few moments. The reason to use StringBuffer is exactly that - it is mutable and allows me to clear it when required. The new version allocates it once and then just modifies it according to need.